core/aws-core/src/test/java/software/amazon/awssdk/awscore/interceptor/FriendlyUnknownHostExceptionInterceptorTest.java

```diff
@@ -0,0 +1,107 @@
+import software.amazon.awssdk.core.internal.interceptor.DefaultFailedExecutionContext;
+import software.amazon.awssdk.regions.Region;
+
+public class FriendlyUnknownHostExceptionInterceptorTest {
+    private static final ExecutionInterceptor INTERCEPTOR = new FriendlyUnknownHostExceptionInterceptor();
+
+    @Test
+    public void modifyException_skipsNonUnknownHostExceptions() {
+        IOException exception = new IOException();
+        assertThat(modifyException(exception)).isEqualTo(exception);
+    }
+
+    @Test
+    public void modifyException_supportsNestedUnknownHostExceptions() {
+        Exception exception = new UnknownHostException();
+        exception.initCause(new IOException());
+        exception = new IllegalArgumentException(exception);
+        exception = new UnsupportedOperationException(exception);
+
+        assertThat(modifyException(exception, Region.AWS_GLOBAL)).isInstanceOf(SdkClientException.class);
+    }
+
+    @Test
+    public void modifyException_returnsGenericHelp_forGlobalRegions() {
+        UnknownHostException exception = new UnknownHostException();
+        assertThat(modifyException(exception, Region.AWS_GLOBAL))
+            .isInstanceOf(SdkClientException.class)
+            .hasMessageContaining("network");
+    }
+
+    @Test
+    public void modifyException_returnsGenericHelp_forUnknownServices() {
+        UnknownHostException exception = new UnknownHostException();
+        assertThat(modifyException(exception, Region.US_EAST_1, "millems-hotdog-stand"))
+            .isInstanceOf(SdkClientException.class)
+            .hasMessageContaining("network");
+    }
+
+    @Test
+    public void modifyException_returnsGenericHelp_forUnknownServicesInUnknownRegions() {
+        UnknownHostException exception = new UnknownHostException();
+        assertThat(modifyException(exception, Region.of("cn-north-99"), "millems-hotdog-stand"))
+            .isInstanceOf(SdkClientException.class)
+            .hasMessageContaining("network");
+    }
+
+    @Test
+    public void modifyException_returnsGenericHelp_forServicesRegionalizedInAllPartitions() {
+        UnknownHostException exception = new UnknownHostException();
+        assertThat(modifyException(exception, Region.US_EAST_1, "dynamodb"))
+            .isInstanceOf(SdkClientException.class)
+            .hasMessageContaining("network");
+    }
+
+    @Test
+    public void modifyException_returnsGenericGlobalRegionHelp_forServicesGlobalInSomePartitionOtherThanTheClientPartition() {
+        UnknownHostException exception = new UnknownHostException();
+        assertThat(modifyException(exception, Region.of("cn-north-99"), "iam"))
+            .isInstanceOf(SdkClientException.class)
+            .hasMessageContaining("aws-global")
+            .hasMessageContaining("aws-cn-global");
+    }
+
+    @Test
+    public void modifyException_returnsSpecificGlobalRegionHelp_forServicesGlobalInTheClientRegionPartition() {
```

Comment:
Should you also test that the message does NOT contain "aws-global"?

Reply:
Sure! I thought the same thing, but it was more code since there isn't a messageDoesNotContain, so I was worried it would just distract. I can throw it in a function, though, since I agree it's something I shouldn't have been lazy about.